### DISCUSSION TITLE
Error while trying to create a mock

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#2253] [Q&A] [answered]
GM

**NetworkConfig.s.sol**
![image](https://github.com/user-attachments/assets/861233a5-c387-46b5-9e2f-028e20fb70e5)

****

**FundMe.s.sol**
![image](https://github.com/user-attachments/assets/4c47f602-d4a5-49ad-ba70-f4336984b231)

****
**Error**
```[⠒] Compiling...
[⠒] Compiling 3 files with Solc 0.8.25
[⠘] Solc 0.8.25 finished in 424.75ms
Error: 
Compiler run failed:
Error (9582): Member "priceFeedAddress" not found or not visible after argument-dependent lookup in function () view external returns (address).
  --> script/FundMe.s.sol:11:36:
   |
11 |         address priceFeedAddress = networkConfig.thisNetwork.priceFeedAddress;
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

what should i do to pass the error ?

GN

**Comment by AztecWarlord** [comment]
Try using the @ in front of the ```forge-std``` like this ```@forge-std/script.sol```.
to correct the Import issue.

**Reply by MostPow3rful** [reply]
Not Worked. and should not work beacuse before this it worked well
and my remappings :
```
remappings = [
    "src/=src/",
    "script/=script/",
    "@chainlink/=lib/chainlink-brownie-contracts/",
    "forge-std/=lib/forge-std/src/",
]
```

**Comment by EngrPips** [comment]
Hello @MostPow3rful, Please show your `NetworkConfig` file also try `networkConfig.thisNetwork().priceFeedAddress`

**Reply by MostPow3rful** [reply]
i attached the picture of NetworkConfig File . and i tried `networkConfig.thisNetwork().priceFeedAddress`, didn't work

**Reply by EngrPips** [reply]
Hello, you are supposed to be doing `networkConfig.thisNetwork().priceFeedAddress`

**Reply by MostPow3rful** [reply]
i didn't understand.
i wrote this code `NetworkConfig.thisNetwork().priceFeedAddress` in my code but raised an error

**Reply by EngrPips** [reply, marked as answer]
Try this `address priceFeedAddress = networkConfig.thisNetwork()`

**Reply by MostPow3rful** [reply]
it worked . thanks. but now i find new logic error LOL.

**Reply by MostPow3rful** [reply]
a question @EngrPips . why i can't access to `priceFeedAddress` property in `FundMe.s.sol` file?
```js
 struct Network {
        address priceFeedAddress;
    }
```

**Reply by MostPow3rful** [reply]
i mean why i can't run this ?
```js
address priceFeedAddress = networkConfig.thisNetwork().priceFeedAddress();
```
or some code like this to access `priceFeedAddress` ?

**Reply by EngrPips** [reply]
Hello @MostPow3rful, You can't do this `networkConfig.thisNetwork().priceFeedAddress();` because you can't get out what is inside of a struct by a function call to the contract that houses the struct, So you need to get the whole struct first then you can deconstruct it to get out the element of the struct which you need. That is why do  the below 

```js
address priceFeedAddress = networkConfig.thisNetwork()
```

**Reply by MostPow3rful** [reply]
ok. thanks <3

**Reply by EngrPips** [reply]
You are welcome, Friend.
